test/e2e/common/pods.go

```diff
@@ -848,4 +853,115 @@ var _ = framework.KubeDescribe("Pods", func() {
+		podResource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"}
+		testNs := "default"
+		testPodName := "pod-test"
+		testPodImage := "nginx"
+		testPodImage2 := "nginx"
+		testPod := v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: testPodName,
+				Labels: map[string]string{
+					"test-pod-static": "true",
+				},
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  testPodName,
+						Image: testPodImage,
+					},
+				},
+			},
+		}
+		ginkgo.By("creating a Pod with a static label")
+		_, err := f.ClientSet.CoreV1().Pods(testNs).Create(context.TODO(), &testPod, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create Pod")
+
+		ginkgo.By("setting up a watch for the Pod")
+		podWatchTimeoutSeconds := int64(180)
+		podWatch, err := f.ClientSet.CoreV1().Pods(testNs).Watch(context.TODO(), metav1.ListOptions{LabelSelector: "test-pod-static=true", TimeoutSeconds: &podWatchTimeoutSeconds})
+		framework.ExpectNoError(err, "failed to set up watch")
+
+		podWatchChan := podWatch.ResultChan()
+		ginkgo.By("watching for Pod to be ready")
+		for watchEvent := range podWatchChan {
+			podWatchEvent, ok := watchEvent.Object.(*v1.Pod)
+			framework.ExpectEqual(ok, true, "unable to fix type")
+			if podWatchEvent.Status.Phase == "Running" {
+				break
+			}
+		}
+
+		ginkgo.By("patching the Pod with a new Label and updated data")
+		podPatch, err := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"labels": map[string]string{
+					"podtemplate": "patched",
+				},
+			},
+			"spec": map[string]interface{}{
+				"containers": []map[string]interface{}{
+					{
+						"name":  testPodName,
+						"image": testPodImage2,
+					},
+				},
+			},
+		})
+		framework.ExpectNoError(err, "failed to marshal JSON patch for Pod")
+		_, err = f.ClientSet.CoreV1().Pods(testNs).Patch(context.TODO(), testPodName, types.StrategicMergePatchType, []byte(podPatch), metav1.PatchOptions{})
+		framework.ExpectNoError(err, "failed to patch Pod")
+
+		ginkgo.By("getting the Pod and ensure it's patched")
+		pod, err := f.ClientSet.CoreV1().Pods(testNs).Get(context.TODO(), testPodName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "failed to fetch Pod")
+		framework.ExpectEqual(pod.ObjectMeta.Labels["test-pod-static"], "true", "failed to patch Pod - missing label")
+		framework.ExpectEqual(pod.Spec.Containers[0].Image, testPodImage2, "failed to patch Pod - wrong image")
+
+		ginkgo.By("getting the PodStatus")
+		podStatusUnstructured, err := dc.Resource(podResource).Namespace(testNs).Get(context.TODO(), testPodName, metav1.GetOptions{}, "status")
+		framework.ExpectNoError(err, "failed to fetch PodStatus")
+		podStatusUjson, err := json.Marshal(podStatusUnstructured)
+		framework.ExpectNoError(err, "failed to marshal unstructured response")
+		var podStatus v1.Pod
+		json.Unmarshal(podStatusUjson, &podStatus)
```

Comment:
check the unmarshal error to ensure roundtrip:
```
err = json.Unmarshal(podStatusUjson, &podStatus)
framework.ExpectNoError(err, "failed to unmarshal Pod status")
```

Reply:
Updated in 1e98c1ba6c0

Reply:
Updated in 012142769f6